api/handler_test.go

```diff
@@ -145,30 +166,27 @@ func (r *handlerTransport) Submit(op *runtime.ClientOperation) (interface{}, err
 }
 
 func TestServer_BasicAuth(t *testing.T) {
-	handler, deps := getHandler(t, "")
+	clt, _ := NewClient(t, "")
 
 	// create user
-	creds := createDefaultAdminUser(deps.auth, t)
-
-	// setup client
-	clt := client.Default
-	clt.SetTransport(&handlerTransport{Handler: handler})
+	creds := createDefaultAdminUser(clt, t)
+	bauth := httptransport.BasicAuth(creds.AccessKeyID, creds.AccessSecretKey)
 
 	t.Run("valid Auth", func(t *testing.T) {
-		_, err := clt.Repositories.ListRepositories(&repositories.ListRepositoriesParams{}, httptransport.BasicAuth(creds.AccessKeyID, creds.AccessSecretKey))
+		_, err := clt.Repositories.ListRepositories(repositories.NewListRepositoriesParamsWithTimeout(timeout), bauth)
 		if err != nil {
-			t.Fatalf("did not expect error when passing valid credentials")
+			t.Fatalf("unexpected error \"%s\" when passing valid credentials", err)
 		}
 	})
 
 	t.Run("invalid Auth secret", func(t *testing.T) {
-		_, err := clt.Repositories.ListRepositories(&repositories.ListRepositoriesParams{}, httptransport.BasicAuth(creds.AccessKeyID, "foobarbaz"))
+		_, err := clt.Repositories.ListRepositories(repositories.NewListRepositoriesParams().WithTimeout(timeout), httptransport.BasicAuth(creds.AccessKeyID, "foobarbaz"))
 		if err == nil {
 			t.Fatalf("expect error when passing invalid credentials")
 		}
 		errMsg, ok := err.(*repositories.ListRepositoriesUnauthorized)
```

Comment:
not required - but after this call we verify the error, here is a better option
```go
		if !errors.As(err, &unauthorizedErr) {
			t.Fatalf("got %v not default error answer", err)
		} else if !strings.EqualFold(unauthorizedErr.GetPayload().Message, "error authenticating request") {
			t.Fatalf("expected authentication error, got error: %s", unauthorizedErr.GetPayload().Message)
		}
```

Reply:
Cool; done!